docs/DeveloperGuide.md

```diff
@@ -10,6 +10,72 @@ title: Developer Guide
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Implementation of `Command` classes
+
+#### Current Implementation
+There are 4 different subclasses of `Item` in InternHunter, namely `Company`, `Internship`, `Application`, and
+`Profile`. <br/>
+
+There are 2 types of commands:
+- Commands that are dependent on the type of `Item`
+    - e.g. `AddCommand`, `DeleteCommand`, `EditCommand`
+    - These commands are implemented as _abstract_ classes that inherit from the `Command` class. Type specific
+    commands like `AddCompanyCommand` and `AddApplicationCommand` will then inherit from the _abstract_ `AddCommand`
+    class
+- Commands that are not dependent on the type of `Item`
+    - e.g. `HelpCommand`, `ExitCommand`
+    - These commands are implemented as _concrete_ classes and inherit directly from the `Command` class
+
+From this point on, we will be using one particular command that is dependent on the type of `Item`, `AddCommand`
+to facilitate discussion.
+
+The following is an example of the class hierachy:
+
+![CommandClassDiagram](images/CommandClassDiagram.png)
+
+#### Design considerations
+
+##### Aspect: Whether `AddCommand` should be abstract and split into 4 other types of `AddItemCommand` or handle the addition of the 4 `Item` subclasses
+
+**Alternative 1 (current choice)**: `AddCommand` is split into 4 other types of `AddItemCommand`. Parser parses the
+user input and creates the specific `AddItemCommand` for execution. The following activity diagram shows how the 
+execution of the `AddApplicationCommand` will work.
+
+![AddApplicationCommandActivityDiagram](images/AddApplicationCommandActivityDiagram.png)
+
+- Pros: 
+    - Each command has its own specific task to execute. This means that classes are more flexible and can be changed
+    very easily. 
+    - Higher cohesion as the class is only dependent on the one `Item` type
```

Comment:
Why does it have higher cohesion?
Isn't low dependency --> low coupling?

Reply:
Did you misread? 
Quoting the textbook: 
Cohesion is a measure of how strongly-related and focused the various responsibilities of a component are.

Reply:
Coupling is a measure of the degree of dependence between components, classes, methods, etc. Low coupling indicates that a component is less dependent on other components. ?

Reply:
There's no coupling word used here?

Reply:
"Only dependent on the one `Item` type" --> low dependency --> low coupling?